scripts/lib/CIME/build.py

```diff
@@ -215,7 +215,7 @@ def _build_libraries(case, exeroot, sharedpath, caseroot, cimeroot, libroot, lid
         if (not os.path.exists(shared_item)):
             os.makedirs(shared_item)
     mpilib = case.get_value("MPILIB")
-    libs = ["gptl", comp_interface, "pio", "csm_share"]
+    libs = ["gptl", "mct", "pio", "csm_share"]
```

Comment:
This seems odd. @sarich ?

Reply:
Perhaps there more nuance here than I first realized. 

if --driver is used on the create_newcase command line, then COMP_INTERFACE is set to that value. In case.py:
```
self.set_lookup_value("COMP_INTERFACE", driver)
```
That value is used in build.py to build the driver:
```
        config_dir = os.path.join(cimeroot, "src", "drivers", comp_interface, "cime_config")
        run_cmd("{}/buildexe {} {} {} "
                       .format(config_dir, caseroot, libroot, bldroot), 
```

But it is also used to build external library src/externals/mct  in _build_libraries():
```
    libs = ["gptl", comp_interface, "pio", "csm_share"]
```

Now there's a confusion because the external library and the driver have the same name.
Does comp_interface refer to the driver component or the underlying library for communications or are they assumed to be the same? 

The moab driver, in it's fledgling current state, relies on the mct external library. This may get completely phased out.  Do I need to add another library src/externals/moab that is a copy of the mct library?

Link for reference: https://github.com/E3SM-Project/E3SM/pull/2513

Reply:
@jedwards4b , can you comment here?

Reply:
What is the error that you get when you leave the comp_interface variable here?  We do build the mct external library with the nuopc driver and the line as it was without this change.

Reply:
Does the nuopc fork have a buildlib.nuopc file that builds the mct lib?

Reply:
Oops _ i just looked at this code in the nuopc branch and we have the same solution - this is okay.   Sorry for the confusion.